### DISCUSSION TITLE
Use SHM between processes in a machine and use TCP/IP in defferent machines automatically

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3646] [General]
How to code?

**Comment by JLBuenoLopez** [comment]
Hi @wjhtinger,

You can find every information about enabling transports in Fast DDS in the [Transport layer](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/transport.html) documentation section. You should create as many transports as you need and then attached them to the DomainParticipant using the `user_transports` list. Please remember to disable the builtin transports as well with the `use_builtin_transports` flag. You may find [this other question](https://github.com/eProsima/Fast-DDS/discussions/3478) of interest.
